lib/spec/router.sw-spec.js

```diff
@@ -4,43 +4,186 @@ importScripts('/base/lib/router.js');
 describe('Router instances', function () {
   'use strict';
 
-  it('should have defined methods for available verbs', function () {
-    var router = new Router();
-    expect(router.methods).to.be.defined;
-    expect(router.methods.length).to.be.at.least(1);
-    expect(router.methods).to.contain('all');
-    router.methods.forEach(function (verb) {
-      expect(router[verb]).to.be.defined;
+  var router;
+
+  beforeEach(function() {
+    router = new Router();
+  });
+
+  describe('add()', function() {
+    it('should push new route to the stack', function () {
+      var handler = function(){};
+
+      router.add('get', '/', handler);
+      expect(router.stack[0]).to.deep.equal({
+        method: 'get',
+        path: new RegExp('/'),
+        handler: handler
+      });
+    });
+
+    it('should push a route with "all" methods', function() {
+      var handler = function(){};
+      router.add('all', '/', handler);
+      expect(router.stack[0].method).to.equal('all');
+    });
+
+    it('should lowercase the method name', function () {
+      router.add('GET', '/', function(){});
+      expect(router.stack[0].method).to.equal('get');
+    });
+
+    it('should throw Error when a method is not supported', function () {
+      expect(function() { router.add('invalidMethod') }).to.throw(Error, 'Method "invalidmethod" is not supported');
+    });
+
+    it('should throw Error when an extra space is provided', function () {
```

Comment:
Quite specific. Maybe we should normalize the method name by trimming it as well, I think this is a bug more than an expectative. @arcturus ?